libgadget/blackhole.c

```diff
@@ -600,10 +600,31 @@ blackhole(const ActiveParticles * act, double atime, Cosmology * CP, ForceTree *
         return;
     }
 
+    /* Move the working set high so we can keep the tree we build after making the list and still free this active set.*/
+    ActiveBlackHoles = mymalloc2("activeBH", NumActiveBlackHoles * sizeof(int));
```

Comment:
It isn't exactly - it is used as the argument to all the later treewalks. I keep it in a separate array because it is weird to be passing a subarray of tw_dynfric around.